AnalyticsService/app/routes.py

```diff
@@ -58,3 +62,144 @@ def get_budget_data():
     
     except Exception as e:
         return jsonify({'error': str(e)}), 500
+    
+@routes.route("/analytics/createStripeCustomer", methods=["POST"])
+
+def create_stripe_customer():
+    
+    try:
+        token = request.cookies.get('user_cookie')
+        payload = None
+        
+        payload = decode_jwt(token)
+
+        first_name = payload['first_name']
+        company_id = payload['company_id']
+
+        existing_account_check = StripeAccount.query.filter_by(company_id=company_id).first()
+
+        if not existing_account_check:
+
+            customer = stripe.Customer.create(
+                name=first_name
+            )
+
+            print(customer)
+            customer_id=customer.stripe_id
+
+            new_account = StripeAccount(
+                company_id=company_id,
+                customer_id=customer_id
+            ) 
+
+            db.session.add(new_account)
+            db.session.commit()
+
+            return jsonify({"success": True, "message": "Stripe customer created successfully"}), 200
+        
+        else:
+            return jsonify({"success": False, "error": "Stripe Customer already exists"}), 200
+        
+    except StripeError as e:
+                print('Error creating Stripe customer:', str(e))
+                return jsonify({"success": False, "error": str(e)}), 500
+    
+@routes.route("/analytics/createFinancialConnectionsSession", methods=["POST"])
+
+def create_financial_connections_session():
+
+    try:
+        token = request.cookies.get('user_cookie')
+        payload = None
+
+        payload = decode_jwt(token)
+        company_id = payload['company_id']
+
+        customer_id = StripeAccount.query.filter_by(company_id=company_id).first().customer_id
+
+        print("customer id:", customer_id)
+
+        session = stripe.financial_connections.Session.create(
+            account_holder={"type": "customer", "customer": customer_id},
+            permissions=["balances", "ownership", "payment_method", "transactions"]
+        )  
+
+        print(session)
+
+        client_secret = session.client_secret
+        
+        return jsonify(client_secret), 200
+
+    except Exception as e:
+        print(e)
+        return jsonify({"error": str(e)}), 500
+
+@routes.route("/analytics/stripeAccountID", methods=["POST", "GET"]) 
+
+def store_stripe_account_id():
+     
+    try:
```

Comment:
looks like we got some repeated code, simplify this to a single function to avoid repitition

Reply:
I think both methods are necessary because one creates a stripe customer during a rerender, and the createFinancialConnectionsSession method is triggered when a user presses the "connect your bank account" button in analytics.tsx. lemme know if you have any other questions!